CONTRIBUTING.md

```diff
@@ -0,0 +1,41 @@
+# Coding guidelines
```

Comment:
I'd rather avoid bouncing people around all sorts of links - how about until ``CONTRIBUTING.md`` gets too big, we keep it in one file?